ts/src/hyperliquid.ts

```diff
@@ -0,0 +1,1900 @@
+        //         "prevDayPx": "2381.5"
+        //     }
+        //
+        const quoteId = 'USD';
+        const base = this.safeString (market, 'name');
+        const quote = this.safeCurrencyCode (quoteId);
+        const baseId = this.safeString (market, 'baseId');
+        const settleId = 'USDC';
+        const settle = this.safeCurrencyCode (settleId);
+        let symbol = base + '/' + quote;
+        const contract = true;
+        const swap = true;
+        if (contract) {
+            if (swap) {
+                symbol = symbol + ':' + settle;
+            }
+        }
+        return {
+            'id': undefined,
+            'symbol': symbol,
+            'base': base,
+            'quote': quote,
+            'settle': settle,
+            'baseId': baseId,
+            'quoteId': quoteId,
+            'settleId': settleId,
+            'type': 'swap',
+            'spot': false,
+            'margin': undefined,
+            'swap': swap,
+            'future': false,
+            'option': false,
+            'active': true,
+            'contract': contract,
+            'linear': true,
+            'inverse': false,
+            'taker': undefined,
+            'maker': undefined,
+            'contractSize': undefined,
+            'expiry': undefined,
+            'expiryDatetime': undefined,
+            'strike': undefined,
+            'optionType': undefined,
+            'precision': {
+                'amount': 0.00000001,
+                'price': 0.00000001,
+            },
+            'limits': {
+                'leverage': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'amount': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'price': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'cost': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+            },
+            'created': undefined,
+            'info': market,
+        };
+    }
+
+    async fetchBalance (params = {}): Promise<Balances> {
+        /**
+         * @method
+         * @name hyperliquid#fetchBalance
+         * @description query for balance and get the amount of funds available for trading or funds locked in orders
+         * @see https://hyperliquid.gitbook.io/hyperliquid-docs/for-developers/api/info-endpoint#retrieve-a-users-state
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {string} [params.user] *required* Onchain address in 42-character hexadecimal format; e.g. 0x0000000000000000000000000000000000000000
+         * @returns {object} a [balance structure]{@link https://docs.ccxt.com/#/?id=balance-structure}
+         */
+        const user = this.safeString (params, 'user');
```

Comment:
we should use `handleOptionAndParams` to allow the user to set this parameter as a global setting (and avoid having to provide it on every single call)